src/signing_key.rs

```diff
@@ -106,6 +110,55 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> From<PrivateKeyInfo<'a>> for SigningKey {
+    fn from(pki: PrivateKeyInfo) -> SigningKey {
+        pki.private_key.try_into().expect("Ed25519 private key wasn't 32 bytes")
+    }
+}
```

Comment:
You might want to use `TryFrom` and return an error rather than panicking.

You'll also probably want to validate the algorithm OID and parameters, similar to this:

https://github.com/RustCrypto/utils/blob/92371fd/pkcs8/tests/private_key.rs#L51-L52